core/templates/dev/head/base.html

```diff
@@ -255,10 +255,20 @@
                           <ul ng-if="!windowIsNarrow" class="nav oppia-navbar-tabs">
                             <create-activity-button></create-activity-button>
 
-                            <li class="oppia-clickable-navbar-element pull-right">
-                              <a class="oppia-navbar-tab" href="/donate" translate="I18N_TOPNAV_DONATE">
+                          <ul ng-if="windowIsNarrow" class="nav oppia-navbar-tabs-narrow">
```

Comment:
Yes, that was intentional. The insertion of the "Donation" button within the ul element allows responsiveness to narrow screen widths. If the viewer is on mobile, for example, only the "Sign In" button is shown, while the other buttons disappear.